daemons/fenced/fence_watchdog.in

```diff
@@ -0,0 +1,283 @@
+        sys.exit(0)
+
+    if "-V" in options:
+        print(AGENT_VERSION)
+        sys.exit(0)
+
+
+def sorted_options(avail_opt):
+    """ Return a list of all options, in their internally specified order. """
+
+    sorted_list = [(key, ALL_OPT[key]) for key in avail_opt]
+    sorted_list.sort(key=lambda x: x[1]["order"])
+    return sorted_list
+
+
+def usage(avail_opt):
+    """ Print a usage message. """
+    print(LONG_DESC)
+    print()
+    print("Usage:")
+    print("\t" + agent() + " [options]")
+    print("Options:")
+
+    for dummy, value in sorted_options(avail_opt):
+        if len(value["help"]) != 0:
+            print("   " + value["help"])
+
+
+def metadata(avail_opt, options):
+    """ Print agent metadata. """
+
+    print("""<?xml version="1.0" ?>
+<resource-agent name="%s" shortdesc="%s">
+<longdesc>%s</longdesc>
+<parameters>""" % (agent(), SHORT_DESC, LONG_DESC))
+
+    for option, dummy in sorted_options(avail_opt):
+        if "shortdesc" in ALL_OPT[option]:
+            print('    <parameter name="' + option +
+                  'required="' + ALL_OPT[option]["required"] + '">')
+
+            default = ""
+            default_name_arg = "-" + ALL_OPT[option]["getopt"][:-1]
+            default_name_no_arg = "-" + ALL_OPT[option]["getopt"]
+
+            if "default" in ALL_OPT[option]:
+                default = 'default="%s"' % str(ALL_OPT[option]["default"])
+            elif default_name_arg in options:
+                if options[default_name_arg]:
+                    try:
+                        default = 'default="%s"' % options[default_name_arg]
+                    except TypeError:
+                        ## @todo/@note: Currently there is no clean way how to handle lists
+                        ## we can create a string from it but we can't set it on command line
+                        default = 'default="%s"' % str(options[default_name_arg])
+            elif default_name_no_arg in options:
+                default = 'default="true"'
+
+            mixed = ALL_OPT[option]["help"]
+            ## split it between option and help text
+            res = re.compile(r"^(.*--\S+)\s+", re.IGNORECASE | re.S).search(mixed)
+            if None != res:
+                mixed = res.group(1)
+            mixed = mixed.replace("<", "&lt;").replace(">", "&gt;")
+            print('      <getopt mixed="' + mixed + '" />')
+
+            if ALL_OPT[option]["getopt"].count(":") > 0:
+                print('      <content type="string" ' + default + ' />')
+            else:
+                print('      <content type="boolean" ' + default + ' />')
+
+            print('      <shortdesc lang="en">' + ALL_OPT[option]["shortdesc"] + '</shortdesc>')
+            print('    </parameter>')
+
+    print('  </parameters>\n <actions>')
+    print('    <action name="on" />')
+    print('    <action name="off" />')
+    print('    <action name="reboot" />')
+    print('    <action name="status" />')
```

Comment:
Just noticed this should be "list"

Reply:
hmm ... you mean an additional list action?
Interesting I'm sure I had it.
Might there have been a reason why I removed it?
Maybe to enforce the static-list to be used. But that takes precedence anyway and
I'm misusing it a bit internally as ipc between controld and fenced.
Readded it - will give it a test-run to see if it has some side-effects.

Reply:
I mean list instead of status -- I saw where list was implemented, but I didn't see status

Reply:
It would only matter if someone configured pcmk_host_check as dynamic-list or status

Reply:
I see ... removing status ...